src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ConsiderPassingBaseTypesAsParameters.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
+{
+    using static MicrosoftCodeQualityAnalyzersResources;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class ConsiderPassingBaseTypesAsParameters : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1011";
+
+        internal static readonly DiagnosticDescriptor Rule =
+            DiagnosticDescriptorHelper.Create(
+                RuleId,
+                CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersTitle)),
+                CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersMessage)),
+                DiagnosticCategory.Design,
+                RuleLevel.IdeSuggestion,
+                description: CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersDescription)),
+                isPortedFxCopRule: true,
+                isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var eventArgsTypeSymbol = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemEventArgs);
+
+                context.RegisterOperationBlockStartAction(context => AnalyzeOperationBlockStart(context, eventArgsTypeSymbol));
+            });
+        }
+
+        private static void AnalyzeOperationBlockStart(OperationBlockStartAnalysisContext context, INamedTypeSymbol? eventArgsTypeSymbol)
+        {
+            if (context.OwningSymbol is not IMethodSymbol methodSymbol ||
+                methodSymbol.IsOverride ||
+                methodSymbol.Parameters.Length == 0 ||
+                methodSymbol.IsImplementationOfAnyInterfaceMember() ||
+                methodSymbol.HasEventHandlerSignature(eventArgsTypeSymbol) ||
+                !context.Options.MatchesConfiguredVisibility(Rule, methodSymbol, context.Compilation,
+                    defaultRequiredVisibility: SymbolVisibilityGroup.All) ||
```

Comment:
Lets keep parity with FxCop CA1011 implementation and only report on public APIs by default
```suggestion
                    defaultRequiredVisibility: SymbolVisibilityGroup.Public) ||
```